trs_filer/ga4gh/trs/endpoints/registerObjects.py

```diff
@@ -0,0 +1,73 @@
+"""Controller for adding new objects."""
+
+from random import choice
+from typing import Dict
+from trs_filer.app import logger
+from flask import (current_app, request)
+from pymongo.errors import DuplicateKeyError
+
+
+class CreateToolPostObject:
+    def __init__(self, request: request):
+        self.db_collection = (
+            current_app.config['FOCA'].db.dbs['trsStore'].
+            collections['objects'].client
+        )
+        self.tool_data = request.json
+        self.id_charset = current_app.config['FOCA'].\
+            endpoints['objects']['id_charset']
+        self.id_length = current_app.config['FOCA'].\
+            endpoints['objects']['id_length']
+        self.host_name = current_app.config['FOCA'].server.host
+
+    def create_id(self, charset, length) -> str:
+        """Creates random ID."""
+        return ''.join(choice(charset) for __ in range(length))
+
+    def create_tool_class(self):
+        """Create tool class."""
+        self.tool_data['toolclass'] = {
+            "description": "Temporary tool class.",
+            "id": "123456",
+            "name": "ToolClass"
+        }
+
+    def create_object(self) -> Dict:
+        """
+        Add new tool post objects to TRS Registry.
```

Comment:
Too complicated! `Register tool.` or `Register tool with TRS.` will be clearer and more concise.